### ISSUE TITLE
`require-input-label` Custom label component false positive

### ISSUE THREAD

**Issue by Windvis** [ember-template-lint/ember-template-lint#1965]
I noticed the `require-input-label` triggers an error in some cases when you use a custom label component, instead of the html element:

```hbs
<CustomLabel>
  <input />
</CustomLabel>
```

Maybe these cases can be covered by adding a list of allowed wrapper components as a configuration option for the `require-input-label` rule?

It's possible to work around it by simply adding the for / id combo anyway, so not a major issue.

**Comment by MelSumner** [comment]
I don't think I can think of a way to filter for that (how could I differentiate between a custom label element and any other element?); I think that using the inline disable of the rule would be the best option in this case. 

If you have any ideas though, I'm all ears!

**Comment by Windvis** [comment]
@MelSumner I think it should be possible with an allow list similar to how [`no-implicit-this` works](https://github.com/ember-template-lint/ember-template-lint/blob/master/docs/rule/no-implicit-this.md#configuration) for example?

```js
module.exports = {
  extends: 'recommended',

  'require-input-label: {
    allow: ['CustomLabel']
  },
};
```
